mixnet/client/src/receiver.rs

```diff
@@ -1,42 +1,69 @@
-use std::{error::Error, net::SocketAddr};
+use std::{error::Error, net::SocketAddr, sync::Arc};
 
 use futures::{stream, Stream, StreamExt};
 use mixnet_protocol::Body;
+use mixnet_util::ConnectionCache;
 use nym_sphinx::{
     chunking::{fragment::Fragment, reconstruction::MessageReconstructor},
     message::{NymMessage, PaddedMessage},
     Payload,
 };
-use tokio::net::TcpStream;
+use tokio::{net::TcpStream, sync::Mutex};
 
 use crate::MixnetClientError;
 
 // Receiver accepts TCP connections to receive incoming payloads from the Mixnet.
-pub struct Receiver;
+pub struct Receiver {
+    cache: ConnectionCache,
```

Comment:
For now, we assume that the mixclient listens messages from a single mixnode (that might be operated by the same guy). Then, I don't think we need cache or something for the `Receiver`.